ivadomed/inference.py

```diff
@@ -278,7 +280,19 @@ def segment_volume(folder_model, fname_image, gpu_number=0, options=None):
                                        bin_thr=-1,
                                        postprocessing=context['postprocessing'])
 
-    return pred_nib
+                pred_list = split_classes(pred_nib)
+                target_list = context['loader_parameters']['target_suffix']
+
+    return pred_list, target_list
+
+
+def split_classes(nib_prediction):
+    pred = nib_prediction.get_fdata()
+    pred_list = []
+    for c in range(pred.shape[-1]):
+        class_pred = nib.Nifti1Image(pred[..., c].astype('float32'), nib_prediction.affine)
```

Comment:
Is `float32` needed / desired here?
Also: I am not sure: are the nibabel files saved this way proper 3D objects? or 4D with empty last channel?
Happy to check, let me know :-)

Reply:
I added this to comply to this test: https://github.com/ivadomed/ivadomed/blob/master/testing/test_segment_volume.py#L139
Or else it is in float64 which I believe is overkill.

3D images can directly be saved with nibabel :), but a 4D with the last dimension to identify channels works as well.

Reply:
Sick!